src/lib/operators/difference.cpp

```diff
@@ -141,7 +141,14 @@ std::shared_ptr<const Table> Difference::_on_execute() {
 
     // Only add chunk if it would contain any tuples
     if (!output_segments.empty() && output_segments[0]->size() > 0) {
-      output_chunks.emplace_back(std::make_shared<Chunk>(output_segments));
+      // Set ordered_by flag
+      const auto chunk = std::make_shared<Chunk>(output_segments);
+      const auto ordered_by = in_chunk->ordered_by();
+      if (ordered_by) {
+        chunk->finalize();
```

Comment:
Our idea behind this: If `ordered_by` is set for a chunk, we finalize it to ensure that the ordering is not mixed up afterwards

Reply:
Just talked to @Bensk1 about this. Marking the chunk as immutable makes, but first we have been afraid of the MVCC overhead.
But since we are creating Reference Segments here, we probably do not execute any code apart from setting the flag (cf. https://github.com/hyrise/hyrise/blob/c4d0cd29ed7260786ea52b4e3d35494d291ac84d/src/lib/storage/chunk.cpp#L95). This should be ensured through the constructor which complaints for Ref segments with MVCC data.

If we do not want to go this path, we see two ways to proceed: (i) having a second constructor that accepts the mutable flag and sets it or (ii) setting the mutable flag in the constructur based whenever we see a reference segment. Any opinions on that?

Reply:
I didn't realize that these chunks do not have MVCC information, so the only thing `finalize()` does is to set `_is_mutable`. Please ignore.